dmoj/checkers/linematches.py

```diff
@@ -0,0 +1,24 @@
+from re import split as resplit
+from typing import List, Union
+
+from dmoj.result import CheckerResult
+from dmoj.utils.unicode import utf8bytes
+
+
+def check(
+    process_output: bytes, judge_output: bytes, point_value: float = 1,
+    point_distribution: List[int] = [1], filler_lines_required: bool = True, **kwargs
+) -> Union[CheckerResult, bool]:
+    process_lines = list(filter(None, resplit(b'[\r\n]', utf8bytes(process_output))))
+    judge_lines = list(filter(None, resplit(b'[\r\n]', utf8bytes(judge_output))))
+
+    if filler_lines_required and len(process_lines) != len(judge_lines):
+        return False
+
+    points = 0
+
+    for i, (process_line, judge_line) in enumerate(zip(process_lines, judge_lines)):
+        if process_line == judge_line:
+            points += point_distribution[i] if i < len(point_distribution) else 0
```

Comment:
Semantically speaking, shouldn't this be WA if `len(point_distributions) != len(process_lines)`? Can `len(judge_lines) != len(point_distribution)`? What would that mean?

Reply:
It's possible for `len(point_distributions) != len(process_lines)` to still be AC (get partial points) as there are some problems which don't require the user to output the complete number of lines, 
see https://dmoj.ca/problem/coci11c2p1 - "even if the second line of output is not present".

Ideally, `len(point_distributions)` and `len(judge_lines)` should always be equal, which would be the responsibility of the user configuring the problem yml. Should it just assume they are equal and IE if the problem creator fails to provide a valid point distribution?

There's also the issue where if the `point_distribution` passed is `[]` or `[0]` in which case the checker would just die from division by zero.

Reply:
Let's fail if the distribution and number of judge lines are not matched, or if we'd end up dividing by zero. A helpful error message would be, well, more helpful to a problemsetter than a random Python exception.

Reply:
Sounds good, changes have been made.